woocommerce/woocommerce-ios
# Dynamic Dashboard: Handle hiding cards from the ellipsis menu

Closes: #12507 


## Description

This PR handles the loading and hiding of dashboard cards to migrate away from the use of UserDefaults. Changes include:
- Added `canShowOnDashboard` for onboarding and Blaze cards to display them as unavailable if necessary.
- Added callback closures for all dashboard card views to update the cards' visibility when dismissing them from the ellipsis menu.
- Updated the layout of the onboarding card to be consistent with the new design.

## Testing instructions

- Enable the feature flag `dynamicDashboard` and build the app.
- Log in to a store with incomplete onboarding tasks. 
- Enable the onboarding card and confirm that the layout of the card is correct.
- Hide any of the card on the dashboard screen, confirm that they disappear from the screen afterward.
- Tap the Settings icon on the top right of the dashboard screen, confirm that you can enable any card to make them show up again.
- Switch to a store eligible for Blaze or completed all onboarding tasks.
- Tap the Settings icon on the top right of the dashboard screen, confirm that either Blaze or onboarding card is marked as unavailable.

Optional: disable the `dynamicDashboard` feature flag and confirm that hiding the Blaze and onboarding cards still works.

## Screenshots


<img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/f803a778-7e83-44ca-84d2-e95fa81762db" width=320 />


---
